### PR TITLE
Property Collector updates should be triggered on the empty FilterSet

### LINKED ISSUE #3631: [BUG] Property Collector Updates should be triggered on the empty FilterSet
**Describe the bug**
The `PropertyCollector` update closure passed to `WaitForUpdatesEx` is only triggered when a non-empty `FilterSet` occurs. This is an issue when side effects need to be triggered inside the update function such as notification mechanisms that an update was acknowledged.

It should be the choice of the update closure to determine whether to process the empty `FilterSet` or not.

**To Reproduce**

1. Create a container view with no child entities such as an empty vm or host folder.
2. Call wait for updates on the view.
3. Observe that the update closure passed to WFU was not triggered even when an iteration of the WFU loop was processed.

**Expected behavior**
The `onUpdatesFn` should be triggered even if the `FilterSet` is empty.

**Affected version**
Since 95aa2571dac212bfb067147a2bcf85b0eeb9afa6.

**Screenshots/Debug Output**
N/A

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by brakthehack**
The PropertyCollector update closure passed to WaitForUpdatesEx is only triggered when a non-empty FilterSet occurs.
This is an issue when side effects need to be triggered inside the update function such as notification mechanisms that an update was acknowledged.

It should be the choice of the update closure to determine whether to process the empty FilterSet or not.

Closes #3631

## Description

Please include a summary of the change.

Closes: #(issue-number)

## How Has This Been Tested?

Please describe any manual tests done to verify your changes.

## Guidelines

Please read and follow the `CONTRIBUTION` [guidelines] of this project.

[guidelines]: https://github.com/vmware/govmomi/blob/main/CONTRIBUTING.md

### CODE REVIEW

File: property/example_test.go [hunk 1]
```diff
@@ -31,7 +32,7 @@ import (
 )
 
 // Example to retrieve properties from a single object
-func ExampleCollector_RetrieveOne() {
+func TestExampleCollector_RetrieveOne(*testing.T) {
```
Comment (dougm) [inline property/example_test.go:35]: These are [Testable Examples](https://go.dev/blog/examples) that are included in the godoc.  Please revert the use of `testing.T` in this file.

The new emptyUpdateSet test can be added to collector_test.go where we using `testing.T`

**Review by dougm — COMMENTED** [review]
Thanks @brakthehack , WaitForUpdatesEx change lgtm. See comments on the test changes.

We also have a DCO bot now, can you [sign](https://github.com/vmware/govmomi/blob/main/CONTRIBUTING.md#sign-off-your-work) your commit?

File: property/example_test.go (hunk 1 above)
Comment (brakthehack) [reply property/example_test.go:35]: Thanks, will do.

**Review by dougm — APPROVED** [review]
lgtm, thanks @brakthehack !

**Pull request merged**